### DISCUSSION TITLE
[New feature] Only Component Towers can integrate with walls; expand this

### DISCUSSION THREAD

**Discussion by askhati** [Vinifera-Developers/Vinifera#1480] [Feature Requests]
### Description:
In TS, some buildings can integrate with overlay walls. This is commonly used by GDI and their Component Towers: once a segment of wall is placed down, a Component Tower can be placed on top of the wall (which automatically removes the underlying wall overlay) and the wall will automatically integrate with the wall, i.e. treating it as another wall cell. This functionality, where the game recognizes some buildings as wall-integrators, is apparently hardcoded for the GDI Component Tower (object code GACTWR), and there is no option to allow other buildings to also integrate with walls in the same manner.

### Possible Implementation:
New rules.ini code tag: "LinkWithWall=y/n" (only works on structures)
If set to True, this building interacts with walls in the same way as the vanilla Component Tower does (can be placed over existing walls, and existing walls will link up with it).

**Comment by Metadorius** [comment]
Not sure why it's labeled as vanilla bug in title. Otherwise it's fairly simple to do.

**Comment by askhati** [comment]
Just to clarify: is it already possible & simple to do this in the vanilla build, or would it be a simple edit for Vinifera?

**Comment by Metadorius** [comment]
The latter, pretty simple to code that in Vinifera, as well as Phobos so to say.
